lib/cppcheck.cpp

```diff
@@ -1014,16 +1014,37 @@ unsigned int CppCheck::checkFile(const FileWithDetails& file, const std::string
 
         // Get configurations..
         std::set<std::string> configurations;
-        if ((mSettings.checkAllConfigurations && mSettings.userDefines.empty()) || mSettings.force) {
-            Timer::run("Preprocessor::getConfigs", mSettings.showtime, &s_timerResults, [&]() {
-                configurations = preprocessor.getConfigs(tokens1);
-            });
+        Timer::run("Preprocessor::getConfigs", mSettings.showtime, &s_timerResults, [&]() {
```

Comment:
This code will always execute `Preprocessor::getConfigs` even if max configs is 1.